src/credentials/FabricTable.h

```diff
@@ -423,6 +478,90 @@ class DLL_EXPORT FabricTable
      */
     CHIP_ERROR StoreFabricIndexInfo() const;
 
+    static constexpr size_t LastKnownGoodTimeTLVMaxSize()
+    {
+        // We have Last Known Good Time and, optionally, a previous Last known
+        // good time for fail safe cleanup.
+        return TLV::EstimateStructOverhead(sizeof(uint32_t), sizeof(uint32_t));
+    }
+
+    /**
+     * Initialize Last Known Good Time to the later of firmware build time or
+     * last known good time persisted in storage.  Persist the selected value
+     * if this differs from the value in storage or no value is yet persisted.
+     *
+     * @return CHIP_NO_ERROR on success, else an appropriate CHIP_ERROR
+     */
+    CHIP_ERROR InitLastKnownGoodChipEpochTime();
+
+    /**
```

Comment:
Please move *all* of this API to a new class. None of this truly relates to the fabric table

Reply:
Last Known Good Time is only mandated to come from firmware build time and certificates installed at commissioning time.

It seems a pretty big stretch then to argue this does not relate to the fabric table.

I'm imagining a compromise though that encapsulates this in its own class, but that that class would be a private member of the FabricTable.

Does that seem reasonable?  If not, can you suggest an architecture that would be preferred?  This logic either needs to:

* be inside the fabric table
* be inside the fabric table as a private member class
* needs to itself consume a fabric table reference

My point is that either Last Known Good Time needs to know about the Fabric Table, or the Fabric Table needs to know about Last Known Good Time.  I'm also trying to avoid an implementation that needs to do friend class things.

Reply:
Discussed offline.  Will encapsulate Last Known Good Time as a separate class and have this as a member of FabricTable.

Reply:
Done.